error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -1579,6 +1590,107 @@ Mono<ImmutableSet<T>> after(Flux<T> flux) {
     }
   }
 
+  /** Prefer {@link Flux#sort()} over more verbose alternatives. */
+  static final class FluxSort<T extends Comparable<? super T>> {
+    @BeforeTemplate
+    Flux<T> before(Flux<T> flux) {
+      return flux.sort(naturalOrder());
+    }
+
+    @AfterTemplate
+    Flux<T> after(Flux<T> flux) {
+      return flux.sort();
+    }
+  }
+
+  /** Prefer {@link MathFlux#min(Publisher)} over less efficient alternatives. */
+  static final class FluxTransformMin<T extends Comparable<? super T>> {
+    @BeforeTemplate
+    Mono<T> before(Flux<T> flux) {
+      return flux.sort().next();
+    }
+
+    @AfterTemplate
+    Mono<T> after(Flux<T> flux) {
+      return flux.transform(MathFlux::min).next();
+    }
+  }
+
+  /**
+   * Prefer {@link MathFlux#min(Publisher, Comparator)} over less efficient or more verbose
+   * alternatives.
+   */
+  static final class FluxTransformMinWithComparator<T extends Comparable<? super T>> {
+    @BeforeTemplate
+    Mono<T> before(Flux<T> flux, Comparator<? super T> cmp) {
+      return Refaster.anyOf(
+          flux.sort(cmp).next(), flux.collect(minBy(cmp)).flatMap(Mono::justOrEmpty));
+    }
+
+    @AfterTemplate
+    Mono<T> after(Flux<T> flux, Comparator<? super T> cmp) {
+      return flux.transform(f -> MathFlux.min(f, cmp)).next();
+    }
+  }
+
+  /** Prefer {@link MathFlux#max(Publisher)} over less efficient alternatives. */
+  static final class FluxTransformMax<T extends Comparable<? super T>> {
+    @BeforeTemplate
+    Mono<T> before(Flux<T> flux) {
+      return flux.sort().last();
+    }
+
+    @AfterTemplate
+    Mono<T> after(Flux<T> flux) {
+      return flux.transform(MathFlux::max).next();
```

Comment:
Food for thought: `MathFlux#max` returns a `Mono`, so the resulting `Flux` has either zero or one element. Should we make this more explicit but using `Flux#singleOrEmpty()`? This comment applies for more parts of the PR.

```suggestion
      return flux.transform(MathFlux::max).singleOrEmpty();
```

Reply:
Nice suggestion; applied!

Reply:
Maybe this opens up a box of other possible improvements.

Any conversion `Mono -> Flux -> Mono` for which we know at most one element emitted and that's currently using `Flux#next` should be rewritten? E.g. `flux.transform(xyz -> mono).next()` could call `Flux#singleOrEmpty()`.